### PR TITLE
[DO-1580][minor] filtering of quality investigations

### PR DESCRIPTION
**Pull request by JoelDietz**
feature: Added the functionality of filtering to the quality investigations.
feature: Added a marker to all filtering to mark the active filters.
feature: Added hints to the plain text filters and the selection filters.
chore: reworked the way filtering is implemented on all tables so that there is less duplication.

### CODE REVIEW

File: frontend/src/app/modules/page/alerts/presentation/alerts.component.html [hunk 1]
```diff
@@ -73,17 +76,17 @@ <h3 class="pl-2 mb-4 flex items-center tableHeader">
...
       (onQueuedAndRequestedTableConfigChanged)="onQueuedAndRequestedTableConfigChange($event)"
       (selected)="openDetailPage($event)"
-      [menuActionsConfig]="menuActionsConfig">
+      [menuActionsConfig]="menuActionsConfig"
+    >
       <div emptyContent>
-        <img src="/assets/images/icons/noAlert.svg"
-          class="center py-5"
-          alt="No alert" />
+        <img src="/assets/images/icons/noAlert.svg" class="center py-5" alt="No alert" />
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/page/alerts/presentation/alerts.component.html:86]: We should talk about our prettier settings again, they still don't seem to be aligned

File: frontend/src/app/modules/page/alerts/presentation/alerts.component.ts [hunk 2]
```diff
@@ -61,95 +59,33 @@ export class AlertsComponent {
...
-  ];
-
+  public readonly filterConfigOptions = new FilterCongigOptions();
   public readonly alertsReceivedFilterConfiguration: any[] = [
-    { filterKey: 'createdDate', isTextSearch: false, isDate: true, option: this.optionTextSearch },
-    { filterKey: 'description', isTextSearch: true, option: this.optionTextSearch },
-    { filterKey: 'status', isTextSearch: false, option: this.statusOptions },
-    { filterKey: 'severity', isTextSearch: false, option: this.severityOptions },
-    { filterKey: 'createdBy', isTextSearch: true, option: this.optionTextSearch },
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdDate,
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/page/alerts/presentation/alerts.component.ts:64]: Please use object destructuring here

File: frontend/src/app/modules/page/alerts/presentation/alerts.component.ts [hunk 3]
```diff
@@ -61,95 +59,33 @@ export class AlertsComponent {
...
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdBy,
   ];
 
   public readonly alertsQueuedAndRequestedFilterConfiguration: any[] = [
-    { filterKey: 'createdDate', isTextSearch: false, isDate: true, option: this.optionTextSearch },
-    { filterKey: 'description', isTextSearch: true, option: this.optionTextSearch },
-    { filterKey: 'status', isTextSearch: false, option: this.statusOptions },
-    { filterKey: 'severity', isTextSearch: false, option: this.severityOptions },
-    { filterKey: 'sendTo', isTextSearch: true, option: this.optionTextSearch },
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdDate,
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/page/alerts/presentation/alerts.component.ts:72]: Please use object destructuring here

File: frontend/src/app/modules/page/investigations/presentation/investigations.component.ts [hunk 4]
```diff
@@ -52,11 +60,36 @@ export class InvestigationsComponent {
 
   public investigationReceivedSortList: TableHeaderSort[] = [];
   public investigationQueuedAndRequestedSortList: TableHeaderSort[] = [];
-  private ctrlKeyState: boolean = false;
 
+  public filterReceived: TableFilter = { filterMethod: FilterMethod.AND };
+  public filterQueuedAndRequested: TableFilter = { filterMethod: FilterMethod.AND };
+  public readonly filterConfigOptions = new FilterCongigOptions();
+  public readonly investigationsReceivedFilterConfiguration: any[] = [
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdDate,
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/page/investigations/presentation/investigations.component.ts:68]: Please use object destructuring here

File: frontend/src/app/modules/page/investigations/presentation/investigations.component.ts [hunk 5]
```diff
@@ -52,11 +60,36 @@ export class InvestigationsComponent {
...
+    this.filterConfigOptions.filterKeyOptionsNotifications.description,
+    this.filterConfigOptions.filterKeyOptionsNotifications.status(TranslationContext.COMMONINVESTIGATION),
+    this.filterConfigOptions.filterKeyOptionsNotifications.targetDate,
+    this.filterConfigOptions.filterKeyOptionsNotifications.severity,
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdBy,
+  ];
+
+  public readonly investigationsQueuedAndRequestedFilterConfiguration: any[] = [
+    this.filterConfigOptions.filterKeyOptionsNotifications.createdDate,
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/page/investigations/presentation/investigations.component.ts:77]: Please use object destructuring here

File: frontend/src/app/modules/shared/components/table/table.component.ts [hunk 6]
```diff
@@ -356,13 +360,13 @@ export class TableComponent {
...
-    //Should the filtering be reset every time? Else remove the following line:
-    this.filtering = { filterMethod: FilterMethod.AND };
+    //Should the filtering be reset every time? then add the following line:
+    // this.filtering = { filterMethod: FilterMethod.AND };
     let filterAdded: FilterInfo | FilterInfo[];
     const filterSettings = this.tableConfig.filterConfig.filter(filter => filter.filterKey === filterName)[0];
 
     if (filterSettings.option.length > 0 && !isDate) {
-      this.filtering.filterMethod = FilterMethod.OR;
+      // this.filtering.filterMethod = FilterMethod.OR;
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/shared/components/table/table.component.ts:369]: Can this commented code be removed?

File: frontend/src/app/modules/shared/components/table/table.component.ts [hunk 7]
```diff
@@ -373,16 +377,31 @@ export class TableComponent {
         }
       });
       filterAdded = filterOptions;
+      if (filterAdded.length > 0) {
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/shared/components/table/table.component.ts:380]: you can simplify this statement here: 

this.filterActive[filterName] = filterAdded.length > 0;

File: frontend/src/app/modules/shared/components/table/table.component.ts [hunk 8]
```diff
@@ -373,16 +377,31 @@ export class TableComponent {
...
+      } else {
+        this.filterActive[filterName] = false;
+      }
     } else if (isDate) {
       filterAdded = {
         filterValue: this.filterFormGroup.get(filterName).value,
         filterOperator: FilterOperator.AT_LOCAL_DATE,
       };
+      if (filterAdded.filterValue === null) {
+        this.filterActive[filterName] = false;
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/shared/components/table/table.component.ts:391]: This statement can also be simplified

 this.filterActive[filterName] = filterAdded.filterValue !== null;

File: frontend/src/app/modules/shared/components/table/table.component.ts [hunk 9]
```diff
@@ -373,16 +377,31 @@ export class TableComponent {
...
+      } else {
+        this.filterActive[filterName] = true;
+      }
     } else {
       filterAdded = {
         filterValue: this.filterFormGroup.get(filterName).value,
         filterOperator: FilterOperator.STARTS_WITH,
       };
+      if (filterAdded.filterValue === '') {
+        this.filterActive[filterName] = false;
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/shared/components/table/table.component.ts:401]: Please simplify this statement as well

File: frontend/src/app/modules/shared/helper/filter-helper.ts [hunk 10]
```diff
@@ -79,7 +79,7 @@ export function addFilteringParams(filtering: TableFilter, params: HttpParams):
             }
           }
         } else {
-          if (filtering[key].filterValue !== '') {
+          if (filtering[key].filterValue !== '' && filtering[key].filterValue !== null) {
```
Comment (viannbreslerAccso) [inline frontend/src/app/modules/shared/helper/filter-helper.ts:82]: Please use destructuring here: 

const { filterValue } = filtering[key]

**Review by viannbreslerAccso — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/page/investigations/presentation/investigations.component.ts (hunk 4 above)
Comment (JoelDietz) [reply frontend/src/app/modules/page/investigations/presentation/investigations.component.ts]: Done

File: frontend/src/app/modules/page/investigations/presentation/investigations.component.ts (hunk 5 above)
Comment (JoelDietz) [reply frontend/src/app/modules/page/investigations/presentation/investigations.component.ts]: Done

File: frontend/src/app/modules/shared/components/table/table.component.ts (hunk 6 above)
Comment (JoelDietz) [reply frontend/src/app/modules/shared/components/table/table.component.ts]: Done

File: frontend/src/app/modules/shared/components/table/table.component.ts (hunk 7 above)
Comment (JoelDietz) [reply frontend/src/app/modules/shared/components/table/table.component.ts]: Done

File: frontend/src/app/modules/shared/components/table/table.component.ts (hunk 8 above)
Comment (JoelDietz) [reply frontend/src/app/modules/shared/components/table/table.component.ts]: Done

File: frontend/src/app/modules/shared/helper/filter-helper.ts (hunk 10 above)
Comment (JoelDietz) [reply frontend/src/app/modules/shared/helper/filter-helper.ts]: Done

**Review by viannbreslerAccso — APPROVED** [review]

**Pull request merged**